vcs/gitcmd/repo.go

```diff
@@ -962,6 +962,31 @@ func (r *Repository) Committers(opt vcs.CommittersOptions) ([]*vcs.Committer, er
 	return committers, nil
 }
 
+func (r *Repository) ListFiles(at vcs.CommitID) ([]string, error) {
+	r.editLock.RLock()
+	defer r.editLock.RUnlock()
+
+	if at == "" {
+		at = "HEAD"
+	}
+
+	cmd := exec.Command("git", "ls-files", "--with-tree", string(at))
```

Comment:
Before passing it to a git command, you should sanitize/verify `at` by using `checkSpecArgSafety`, as is already the case in other places here.